src/main/java/com/duke/bot/Parser.java

```diff
@@ -0,0 +1,105 @@
+package com.duke.bot;
+
+import java.time.LocalDate;
+
+public class Parser {
+
+    public Parser() {
+    }
+
+    public static String printSuccessMsg(Task task, TaskList tasks) {
+        String output = "Added: " + task.toString() + "\n";
+        output = output.concat(String.format("Now you have %d task(s) on your list.\n\n", tasks.getSize()));
+        return output;
+        //echo(new Scanner(System.in);
+    }
+
+    public static String parse(String userCommand, DukeUi ui, Storage storage, TaskList tasks) throws DukeException {
+
+        switch(userCommand) {
+
+        case "list":
+            storage.saveToFile(tasks.printList());
+            return tasks.printList();
+
+        case "done":
+            int targetIdx = ui.getNextInt() - 1;
+            tasks.getTask(targetIdx).markDone();
+
+            String output = "Good job! I've marked this task as done:\n";
+            output = output.concat(String.format("%d. %s\n\n", targetIdx + 1, tasks.getTask(targetIdx).toString()));
+            ui.setToken("\\p{javaWhitespace}+");
+            storage.saveToFile(tasks.printList());
+            return output;
+
+        case "todo":
+            ui.setToken("\\n");
+            String userInput = ui.getNextLine();
+            if (userInput.equals("")) {
+                ui.resetScanner();
+                throw new DukeException("Uh-oh! Description of todo cannot be empty!");
+            }
+            TodoTask todoTask = TodoTask.createTodoTask(userInput);
+            tasks.addTask(todoTask);
+
+            ui.setToken("\\p{javaWhitespace}+");
+            storage.saveToFile(tasks.printList());
+            return printSuccessMsg(todoTask, tasks);
+
+        case "deadline":
+            ui.setToken("\\s*/by\\s*|\\n");
+            String action = ui.getNext();
+            String dateInput = ui.getNext();
+            LocalDate byDate = LocalDate.parse(dateInput);
+            DeadlineTask deadlineTask = DeadlineTask.createDeadlineTask(action, byDate);
+            tasks.addTask(deadlineTask);
+
+            ui.setToken("\\p{javaWhitespace}+");
+            storage.saveToFile(tasks.printList());
+            return printSuccessMsg(deadlineTask, tasks);
+
+        case "event":
+            ui.setToken("\\s*/at\\s*|\\n");
+            action = ui.getNext();
+            dateInput = ui.getNext();
+            LocalDate atDate = LocalDate.parse(dateInput);
+            EventTask eventTask = EventTask.createEventTask(action, atDate);
+            tasks.addTask(eventTask);
```

Comment:
I am not sure whether my understanding is correct here, if the EventTask, DeadlineTask, etc. are inherited from one common parent and they can be handled by class-casting, these codes in switch-case can be simplified a bit. :) because tasks.addTask() <- this line is invoked repeatedly now.